.travis.yml

```diff
@@ -12,42 +12,81 @@ addons:
 
 jobs:
   fast_finish: true
   include:
-    - stage: Dev+unit-tests
-      script: docker run -it openhabian/rpi-openhabian bash -c 'bats -r -f "dev-." .'
-    - script: docker run -it openhabian/rpi-openhabian bash -c 'sudo bats -r -f "unit-." .'
-    - stage: Openhabian-Setup
+    - stage: Dev & Unit tests
+      script:
+        - sed -i 's/java_arch=32-bit$/java_arch=64-bit/' build-image/openhabian.conf
+        - docker build -t openhabian/generic-openhabian -f Dockerfile.openhabian-x86 .
+        - docker run -it openhabian/generic-openhabian bash -c 'bats -r -f "dev-." .'
+        - docker run -it openhabian/generic-openhabian bash -c 'bats -r -f "unit-." .'
+    - stage: Run unattended openHABian installation
+      env: amd64
+      script:
+        - export HW=x86
+        - sed -i 's/java_arch=32-bit$/java_arch=64-bit/' build-image/openhabian.conf
+        - docker build -t openhabian/${HW}-openhabian -f Dockerfile.openhabian-x86 .
+        - docker run --name "openhabian-${HW}" --privileged -d openhabian/${HW}-openhabian;
+          docker exec -it "openhabian-${HW}" bash -c \
+            'VIRT=$(sudo virt-what); echo -e "\n\e[37mTesting openhabian install on ${VIRT} ...\n"; ./build.bash local-test && mv ~/.profile ~/.bash_profile && /etc/rc.local; OUTPUT=$?; if [ "${VIRT}" != "native HW" ]; then echo -e "\n\e[32mDid not test openhabian install on native HW, ignoring test result.\n"; exit 0; fi; exit $OUTPUT'
+    - env: arm32hf
       script:
-        - docker run --name "openhabian-test" --privileged -d openhabian/rpi-openhabian
-        - docker exec -it "openhabian-test" bash -c \
+        - docker run --rm --privileged multiarch/qemu-user-static:register --reset
+        - export HW=rpi
+        - sed -i 's/java_arch=64-bit$/java_arch=32-bit/' build-image/openhabian.conf
+        - docker build -t openhabian/${HW}-openhabian -f Dockerfile.openhabian-emulated .
+        - docker run --name "openhabian-${HW}" --privileged -d openhabian/${HW}-openhabian;
+          docker exec -it "openhabian-${HW}" bash -c \
             'VIRT=$(sudo virt-what); echo -e "\n\e[37mTesting openhabian install on ${VIRT} ...\n"; ./build.bash local-test && mv ~/.profile ~/.bash_profile && /etc/rc.local; OUTPUT=$?; if [ "${VIRT}" != "native HW" ]; then echo -e "\n\e[32mDid not test openhabian install on native HW, ignoring test result.\n"; exit 0; fi; exit $OUTPUT'
-        - docker commit "openhabian-test" openhabian-installed
-    - stage: Bats
+    - env: arm32hf
       script:
-        docker run --name "bats-inst-tests" --privileged -d openhabian/rpi-openhabian
-        docker exec -it "bats-inst-tests" bash -c 'bats -r -f "installation-." .'
-    - script:
-        docker run --name "bats-dest-tests" --privileged -d openhabian/rpi-openhabian
-        docker exec -it "bats-dest-tests" bash -c 'bats -r -f "destructive-." .'
+        - docker run --rm --privileged multiarch/qemu-user-static:register --reset
+        - export HW=rpi
```

Comment:
same HW as in L57, so the following calls working on the same VM in parallel.... and with a name clash in docker instances?

Reply:
Note Docker is built in there, using different images ... I don't think it clashes, haven't seen any problems. Changed HW for safety, though. Good find.